### ISSUE TITLE
ChannelShuffling preprocessing layer

### ISSUE THREAD

**Issue by innat** [keras-team/keras-cv#123]
ref. https://albumentations.ai/docs/api_reference/augmentations/transforms/#albumentations.augmentations.transforms.ChannelShuffle

ref https://pytorch.org/docs/stable/generated/torch.nn.ChannelShuffle.html

**Comment by LukeWood** [comment]
Any numbers showing the usefulness of this in training?

**Comment by bhack** [comment]
In Pytorch it was not strictly a [vision preprocessing layer](https://pytorch.org/docs/stable/generated/torch.nn.ChannelShuffle.html)

If I remeber correctly it was intrdouced implementing Shufflenets:
https://arxiv.org/abs/1707.01083

**Comment by LukeWood** [comment]
That's fine.  Users can use it however they like.

**Comment by qlzh727** [comment]
https://github.com/keras-team/keras-cv/pull/124
